get_devices/web_app.js

```diff
@@ -0,0 +1,154 @@
+/**
+ *  Copyright 2022 Google LLC
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+
+// Espruino devices publish a UART service by default.
+const nordicUARTService = '6e400001-b5a3-f393-e0a9-e50e24dcca9e';
+// The test service defined in device_code.js
+const testService = '0b30acec-193e-11eb-adc1-0242ac120002';
+// The test characteristic defined in device_code.js
+const testCharacteristic = '0b30afd0-193e-11eb-adc1-0242ac120002';
+
+let gattServer = undefined;
+let btDeviceId = null;
+const urlParam = 'bt_device_id=';
+const btDeviceIdUrlParam = window.location.search.substring(window.location.search.indexOf(urlParam) + urlParam.length).split('&')[0];
```

Comment:
Use the [`URLSearchParams`](https://developer.mozilla.org/en-US/docs/Web/API/URLSearchParams) interface to parse and query the URL parameters.

Reply:
[URLSearchParams](https://developer.mozilla.org/en-US/docs/Web/API/URLSearchParams) doesn't like `+` in URL parameters. I can either leave it as is or convert the ID to hex before passing it to the URL parameter.